server/modules/protocol/MySQL/mariadbbackend/mysql_backend.c

```diff
@@ -775,6 +845,13 @@ gw_read_and_write(DCB *dcb)
 
         read_buffer = tmp;
 
+        discard_master_wait_gtid_result(&read_buffer, proto);
```

Comment:
Same as with the parameters, this code can be moved inside readwritesplit. I think one place to put this would be in the `clientReply` function in `readwritesplit.cc`  around line 1180, before the call to `reply_is_complete`. This is the point where the readwritesplit does some processing on the result set (which really should be in the protocol module) and checks if the resultset is complete.